### PR TITLE
Add agent visualization and KPI to overview

### PR DESCRIPTION
**Pull request by JuanGarriuz**
### Description

Added a dashboard in overview that shows an agent status and alerts
 
### Issues Resolved

[Add agent visualization and rule severity KPI to overview ](https://github.com/wazuh/wazuh-dashboard-plugins/issues/6550)

### Evidence

image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Overview and check dashboard renders the expected data | :black_circle: | :black_circle: | :black_circle: |
| With new Discover data explorer **_enabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :black_circle: | :black_circle: | :black_circle: |
| With new Discover data explorer **_disabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :black_circle: | :black_circle: | :black_circle: |
| Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Overview and check dashboard render the correct data</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With new Discover data explorer enabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With new Discover data explorer disabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly </summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-query.ts [hunk 1]
```diff
@@ -27,6 +31,14 @@ export const getLastAlertsQuery = (
           },
         },
       },
+      {
+        range: {
+          'rule.level': {
+            gte: ruleLevelRange.minRuleLevel,
+            lte: ruleLevelRange.maxRuleLevel || 15,
```
Comment (Desvelao) [inline plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-query.ts:38]: **suggestion**: the `15` value could be a constant instead.

File: plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-stat.tsx [hunk 2]
```diff
@@ -10,22 +16,87 @@ import {
...
 } from '../../../../react-services/error-management';
 
-export function LastAlertsStat() {
+export function LastAlertsStat({ severity }: { severity: string }) {
   const [countLastAlerts, setCountLastAlerts] = useState<number | null>(null);
   const [discoverLocation, setDiscoverLocation] = useState<string>('');
+  const severityLabel = {
+    low: {
+      label: 'Low',
+      color: UI_COLOR_AGENT_STATUS.active,
```
Comment (Desvelao) [inline plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-stat.tsx:25]: **suggestion**: maybe we should define other constants for these colors instead using the same for agent status connection.

File: plugins/main/server/lib/generate-alerts/generate-alerts-script.js [hunk 3]
```diff
@@ -54,7 +61,7 @@ const ruleDescription = [
   'Sample alert 4',
   'Sample alert 5',
 ];
-const ruleMaxLevel = 14;
+const ruleMaxLevel = 15;
```
[the remark this answers is not in the document]
Comment (asteriscos) [reply plugins/main/server/lib/generate-alerts/generate-alerts-script.js:64]: good catch, on it.

File: plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-query.ts (hunk 1 above)
Comment (asteriscos) [reply plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-query.ts:38]: The query should be able to support only a minimum value of a range. I'll fix it.

**Comment by yenienserrano** [comment]
| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Overview and check dashboard renders the expected data | :black_circle: | :black_circle: | :black_circle: |
| With new Discover data explorer **_enabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :black_circle: | :black_circle: | :black_circle: |
| With new Discover data explorer **_disabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :black_circle: | :black_circle: | :black_circle: |
| Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly | :black_circle: | :black_circle: | :black_circle: |


**Details**
<details>
<summary>:black_circle: Go to Overview and check dashboard render the correct data</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With new Discover data explorer enabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With new Discover data explorer disabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly </summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Comment by Tostti** [comment]
| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Overview and check dashboard renders the expected data | :green_circle: | :green_circle:| :green_circle: |
| With new Discover data explorer **_enabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :green_circle: | :green_circle:| :green_circle: |
| With new Discover data explorer **_disabled_**, go to Overview and check dashboard links redirect to discover adding filters correctly | :green_circle: | :green_circle:| :green_circle: |
| Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly | :green_circle: | :green_circle:| :green_circle: |


**Details**
<details>
<summary>:green_circle: Go to Overview and check dashboard render the correct data</summary>

Chrome - :green_circle:
image
image

Firefox - :green_circle:
image
image

Safari - :green_circle:

image
image


</details>

<details>
<summary>:green_circle: With new Discover data explorer enabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

Safari - :green_circle:

image


</details>

<details>
<summary>:green_circle: With new Discover data explorer disabled, go to Overview and check dashboard links redirect to discover adding filters correctly</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

Safari - :green_circle:
image



</details>

<details>
<summary>:green_circle: Go to Overview and check agents visualization label links redirect to Endpoints summary adding filter correctly </summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

Safari - :green_circle:
image

</details>

**Review by Tostti — APPROVED** [review]
Test ✅ 
CR ✅ 

LGTM

**Pull request merged**